src/sl.rs

```diff
@@ -87,22 +86,22 @@ impl Sl {
 
     pub fn current_thread(&self) -> Thread {
         Thread {
-            thd_ptr: Shared::new(unsafe {
-                sl::sl_thd_curr_rs()
-            }).unwrap()
+            thread_id: unsafe {
+                sl::sl_thdid_rs()
+            }
         }
     }
 
     pub fn spawn<F: FnBox(Sl)>(&self, entrypoint: F) -> Thread {
         let boxed_fn = Box::new(FnBoxWrapper {
-            inner: Box::new(entrypoint)
+            inner: Box::new(entrypoint) 
         });
 
         unsafe {
             let thd_ptr = sl::sl_thd_alloc(closure_spawn_wrapper, Box::into_raw(boxed_fn) as *mut c_void);
             assign_thread_data(thd_ptr);
             Thread {
-                thd_ptr: Shared::new(thd_ptr).unwrap()
+                thread_id: (*thd_ptr).thdid,
```

Comment:
Do you need the *?

Reply:
Unsure actually, i didn't test it the other way, assumed you did because sl_thd_alloc returns a * sl_thd